baseplate/lib/timer.py

```diff
@@ -0,0 +1,56 @@
+from enum import Enum
+from enum import unique
+from typing import Callable
+from typing import Optional
+
+from schedule import CancelJob
+
+from baseplate.lib.threaded_scheduler import ThreadedScheduler
+
+
+@unique
+class TimerState(Enum):
+    Running = "Running"
+    Stopped = "Stopped"
+
+
+class Timer:
+    def __init__(
+        self,
+        function: Callable[[], None],
+        interval: float,
+        scheduler_interval: Optional[float] = None,
+        is_repeating: bool = False,
+    ) -> None:
+        self.function = function
+        self.interval = interval
+        self.scheduler_interval = scheduler_interval if scheduler_interval is not None else interval
```

Comment:
having 2 intervals seems confusing, why not just have one?

Reply:
The `scheduler_interval` is the resolution of the scheduler, i.e. how often it checks for pending work. The `interval` is the actual frequency at which we want to execute our command. But thinking about it, it's probably sufficient to make them the same interval in this instance because there's only ever this one job scheduled.